### ISSUE TITLE
Support BoundUsingLocalDeclarations in IOperation and CFG

### ISSUE THREAD

**Issue by chsienki** [dotnet/roslyn#32100]
BoundUsingLocalDeclarations are currently implemented as OperationKind.None, and return the List of local declarations as its children. We should implement this correctly with an Operation node.

We should also support CFG correctly with using declarations. Currently we report the using declaration as OperationKind.None in a single block, and thus don't report the actual underlying block structure generated by lowering. 

[jcouv update:] When fixing this, let's also verify IDE behavior got fixed. For example https://github.com/dotnet/roslyn/issues/36502

**Comment by jcouv** [comment]
Once fixed, please verify this scenario (copied from https://github.com/dotnet/roslyn/issues/33464):

Using-declaration should not trigger RemoveUnusedValues.

I suspect this is a compiler issue, as the `AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer` relies on the control flow graph to determine how variables are used.

![image](https://user-images.githubusercontent.com/12466233/52920672-28bd8a80-32c4-11e9-8f0e-82b51439ca2e.png)

![image](https://user-images.githubusercontent.com/12466233/52920667-180d1480-32c4-11e9-99d5-cc59d339e5ec.png)

**Comment by yyjdelete** [comment]
Maybe the same as #33464.
An quick fix `Inline temporary variable` should not be provided for the first `x1`.
```cs
using var x1 = File.OpenText("test");
var x2 = x1.ReadToEnd();
```
Expect: `Inline temporary variable` should not be provided.
Actual: 
```cs
var x2 = File.OpenText("test").ReadToEnd();
```

**Comment by mavasani** [comment]
https://github.com/dotnet/roslyn/issues/36502 - another report where IOperation/CFG based analyses in IDE is falling over due to this issue. There are now at least 4 known IDE features that give false results due to this missing support.

**Comment by jinujoseph** [comment]
also reported [here](https://developercommunity.visualstudio.com/content/problem/617862/incorrect-warning-disposable-object-created-by-is.html)

**Comment by mavasani** [comment]
Adding more context to @jinujoseph's comment: We have added new CFG/dataflow analysis based dispose analyzers in the IDE in 16.2, and they generate false positives due to this missing support:

```csharp
using System.Diagnostics;

public class Class1
{
    public void M()
    {
        using Process _ = new Process  // Dispose analyzer flags this object creation as not-disposed.
        {
            StartInfo = new ProcessStartInfo()
        };
    }
}
```

**Comment by mavasani** [comment]
I have created https://github.com/dotnet/roslyn/pull/36734 to workaround this issue in couple of IDE analyzers. When the IOperation/CFG support has been added for using declarations, kindly revert the product workarounds in that PR and ensure the added unit tests still pass.

**Comment by chsienki** [comment]
Fixed with #39125
